src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Lower.java

```diff
@@ -2831,47 +2831,55 @@ public void visitTypeCast(JCTypeCast tree) {
      */
     public void visitTypeTest(JCInstanceOf tree) {
         if (tree.expr.type.isPrimitive() || tree.pattern.type.isPrimitive()) {
-            JCExpression exactnessCheck = null;
+            JCStatement sideEffect;
+            JCExpression exactnessCheck;
             JCExpression instanceOfExpr = translate(tree.expr);
 
-            // preserving the side effects of the value
-            VarSymbol dollar_s = new VarSymbol(FINAL | SYNTHETIC,
-                    names.fromString("tmp" + variableIndex++ + this.target.syntheticNameChar()),
-                    types.erasure(tree.expr.type),
-                    currentMethodSym);
-            JCStatement var = make.at(tree.pos())
-                    .VarDef(dollar_s, instanceOfExpr);
-
             if (types.isUnconditionallyExact(tree.expr.type, tree.pattern.type)) {
+                // instanceOfExpr; true
+                sideEffect = make.Exec(instanceOfExpr);
```

Comment:
maybe `evaluatedLHS` or `memoizedLHS`.

Reply:
This is not really a lhs - we are trying to generate "statement; expression", maybe rename this to `prefixStatement`?

Reply:
True. Yes, good name. Go for it.